domain/model/service/service.go

```diff
@@ -542,3 +589,59 @@ func (s *Service) UpdateCredential(
 
 	return s.st.UpdateCredential(ctx, uuid, key)
 }
+
+// getWatchActivatedModelsMapper returns a mapper function that filters change events to
+// include only those associated with activated models.
+func getWatchActivatedModelsMapper(st State) func(ctx context.Context, db database.TxnRunner,
```

Comment:
We should add a comment to this function saying that the subset of changes returned are maintained in the same order that they are received.

This is a pretty big characteristic we need to maintain and test.

Reply:
Sure will add test cases for this too, thanks tom :)